tools/toolchains/arm.py

```diff
@@ -342,9 +342,9 @@ def __init__(self, target, notify=None, macros=None,
                  build_profile=None, build_dir=None):
         ARM.__init__(self, target, notify, macros, build_dir=build_dir,
                      build_profile=build_profile)
-        if not set(("ARM", "uARM")).intersection(set(target.supported_toolchains)):
-            raise NotSupportedException("ARM/uARM compiler support is required for ARM build")
-
+        #check only for ARMC5 because ARM_STD means using ARMC5, and thus supported_toolchains must include ARMC5
+        if not set(("ARMC5",)).intersection(set(target.supported_toolchains)):
```

Comment:
Nit: Since we're only checking for one toolchain now, you can shorten this check:
```suggestion
        if "ARMC5" not in target.supported_toolchains:
```

Reply:
Fixed.